# ANOVA Test

In [2]:
import pandas as pd
#load libraries
from sklearn import datasets
#load digits dataset
iris = datasets.load_iris()

In [4]:
df =pd.DataFrame(iris.data)

In [5]:
df

0    1    2    3
0    5.1  3.5  1.4  0.2
1    4.9  3.0  1.4  0.2
2    4.7  3.2  1.3  0.2
3    4.6  3.1  1.5  0.2
4    5.0  3.6  1.4  0.2
..   ...  ...  ...  ...
145  6.7  3.0  5.2  2.3
146  6.3  2.5  5.0  1.9
147  6.5  3.0  5.2  2.0
148  6.2  3.4  5.4  2.3
149  5.9  3.0  5.1  1.8

[150 rows x 4 columns]

#### Iris dataset has totally 150 rows, where 1st 50 rows belongs to the class setosa, next 50 rows belongs to class versicolor, last 50 rows belongs to class virginica. 

#### Column 1 is sepal length(SL), column 2 is sepal width(SW), column 3 is petal length(PL), column 4 is petal width(PW)

#### For ANOVA test: Ho is μSL=μSW=μPL=μPW, H1 is atleast one of the mean value is different


In [7]:
import scipy.stats as stats
stats.f_oneway(df.iloc[:,0], df.iloc[:,1],df.iloc[:,2],df.iloc[:,3])

F_onewayResult(statistic=482.91531656927964, pvalue=4.660592480454751e-159)

#### From the above output, pvalue is 4.66 exponential raise to the power -159, which is approximately equal to 0

#### So, here p value is 0 and alpha(by default 0.05) value, hence here alpha value is more than p value ---> reject Ho. Therefore, H1 is accepted which means atleast one of the mean value is different.

# Chi-Square test

In [18]:
bike = pd.read_csv("https://raw.githubusercontent.com/Safa1615/BIKE-RENTAL-COUNT/master/day.csv")

In [19]:
bike.head()

instant      dteday  season  yr  mnth  holiday  weekday  workingday  \
0        1  2011-01-01       1   0     1        0        6           0   
1        2  2011-01-02       1   0     1        0        0           0   
2        3  2011-01-03       1   0     1        0        1           1   
3        4  2011-01-04       1   0     1        0        2           1   
4        5  2011-01-05       1   0     1        0        3           1   

   weathersit      temp     atemp       hum  windspeed  casual  registered  \
0           2  0.344167  0.363625  0.805833   0.160446     331         654   
1           2  0.363478  0.353739  0.696087   0.248539     131         670   
2           1  0.196364  0.189405  0.437273   0.248309     120        1229   
3           1  0.200000  0.212122  0.590435   0.160296     108        1454   
4           1  0.226957  0.229270  0.436957   0.186900      82        1518   

    cnt  
0   985  
1   801  
2  1349  
3  1562  
4  1600

In [20]:
categorical_col = ['season', 'yr', 'mnth', 'holiday', 'weekday', 'workingday','weathersit']
print(categorical_col)

['season', 'yr', 'mnth', 'holiday', 'weekday', 'workingday', 'weathersit']


##### we use the crosstab() function to create a contingency table of the two selected variables to work on ‘holiday’ and ‘weathersit’.

In [21]:
chisqt = pd.crosstab(bike.holiday, bike.weathersit, margins=True)
print(chisqt)

weathersit    1    2   3  All
holiday                      
0           448  241  21  710
1            15    6   0   21
All         463  247  21  731


In [22]:
from scipy.stats import chi2_contingency 
import numpy as np
chisqt = pd.crosstab(bike.holiday, bike.weathersit, margins=True)
value = np.array([chisqt.iloc[0][0:5].values,
                  chisqt.iloc[1][0:5].values])
print(chi2_contingency(value)[0:3]) #applying chi2_contingency() function on the table and getting the chi sq-statistic value, p-value and dof value.

(1.0187958822970438, 0.796704036292652, 3)


##### From the above output chi sq- statistic is 1.01, p-value is 0.79, dof=3. 
##### For chi sq test: H0: chi sq. statistic value < crtical value, H1: chi sq. statistic value > crtical value
#### Critical value is found by using chi sq table with the help of alpha value(by default 0.05) and dof value resp. Here, the critical value is 7.815. Therefore, here critical value is greater than chi sq. statistic value--->accept H0--->hence variables 'holiday' & 'weathersit' are independent of each other.